### ISSUE TITLE
Feat(banner): take place on screen, should we add the possibility of hide

### ISSUE THREAD

**Issue by AugustinMauroy** [nodejs/nodejs.org#6292]
## Problem / situation

When I reviewed the pr, I was 'bothered' by the banner, which took up a lot of space. In some doc creation tools such as Docusaurus, Nextra there is a way to hide the banner to maximise the space dedicated to the content you are reading.

This is what the "learn" section looks like when there is a banner.

<img width="1552" alt="Capture d’écran 2024-02-08 à 09 02 02" src="https://github.com/nodejs/nodejs.org/assets/97875033/00e6e7d8-5b27-4edd-aace-64dec13ae32d">

## A possible solution

One possible solution is to put a small cross at the far right of the banner. This cross hides the way, but if there is a new banner, it must be displayed.

[code already done](https://github.com/nodejs/nodejs.org/tree/feat(banner)-hide)

> **NOTE:** to run the website with new design do `turbo serve:redesign`

**Comment by bmuenzenmeyer** [comment]
I think this is a well-thought out concern - and one you even raised in the [Figma 5 months ago](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5?node-id=422:7920&mode=design#551004165)

agree this is standard  functionality for many tools

we need to weigh the cost vs the intent

- we only use the banner for VERY important things, like a security release
- do we do so for every new version? doubtful
- dismissable banners need to [persist, and cache bust via a key](https://github.com/shuding/nextra/blob/nextra%403.0.0-alpha.17/packages/nextra-theme-docs/src/components/banner.tsx)

I doubt we'll get Hayden's input on this as he's moved on - so, what do others think? I'm on the fence about adding it right now - before the site is even live.

**Comment by AugustinMauroy** [comment]
> we only use the banner for VERY important things, like a security release

Yes, I'll use an example to explain the problem.  Let's say we click on the banner. This means that the user is reading the article, so the banner doesn't do anything for him apart from taking up screen space and mental workload.

**Comment by AugustinMauroy** [comment]
why not also display banner on doc page, why not re-think this proposition ?
We can add small cross icon so people on doc see one time the "node interactiv" and then they can read the docs as casual.

**Comment by trivikr** [comment]
This issue was re-opened after a discussion on OpenJS slack at https://openjs-foundation.slack.com/archives/CVAMEJ4UV/p1783822889586969

The latest content of the banner is a link to RenderATL added in https://github.com/nodejs/nodejs.org/pull/8972
It started appearing on undici docs website, when doc-kit was added in https://github.com/nodejs/undici-website/pull/7

Below is a screen recording of how the banner appears in undici docs. It looks like we're forcing it on Node.js users as it appears on all page loads, the green background highlights it and it can't be closed.

<details>
<summary>Screen recording</summary>

https://github.com/user-attachments/assets/16ac2c0c-4233-473c-a753-15c20f3e2880

</details>

If we don't provide a close button, users will find an alternative.

For example, on [Brave Browser](https://brave.com/) there's an option to "Block Elements".
I used it to hide the banner as there was no close button as shown in the screenshots

<details>
<summary>Screenshots</summary>

<img width="988" height="650" alt="Image" src="https://github.com/user-attachments/assets/61cb2b2c-8cda-4617-8188-9052ca0abd52" />

<img width="953" height="650" alt="Image" src="https://github.com/user-attachments/assets/d3691919-df30-499d-9228-1ddcf37a46de" />

</details>

Now Brave will always hide the banner for me including future announcements which may include security releases.
This happened because there was no first-party close button to hide the banner, and I as a viewer chose a third-party alternative.

**Comment by trivikr** [comment]
I also linked this comment on the PR which uses doc-kit on main docs https://github.com/nodejs/node/pull/62045#issuecomment-4966791912
